C/CImPlot/bundled/cmake/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 3.8.0)
+project(cimplot)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+add_library(cimplot SHARED)
+
+target_sources( cimplot PRIVATE ./cimplot/cimplot.cpp)
+target_sources( cimplot PRIVATE ./cimplot/implot/implot.cpp)
+target_sources( cimplot PRIVATE ./cimplot/implot/implot_demo.cpp)
+target_sources( cimplot PRIVATE ./imgui/imgui.cpp)
```

Comment:
If I run this via Cmake, using the imgui sources matched to CImGui.jl (e.g. tag v1.75), I get segfaults when trying to run a demo that uses CImPlot. Downloading binaries from the CImGui_jll repo and setting cimplot to link libcimgui.so however works just fine.

Reply:
Also note: In _this_ PR the hashes are set to v1.76 of imgui in anticipation of the version bump propagating to `CImGui.jl`. When testing locally I'm matching the hashes/tags to be consistent for the current master of `CImGui.jl`, which is still using v1.75.0 of `CImGui_jll.jl`

Reply:
Could you share the demo code used in the test? What version of CImGui_jll is used? Is it v1.75? CImGui.jl still uses CImGui_jll-v1.75 for now, I haven't upgraded it to v1.76 because there are a lot of API changes on the cimgui side(it exposes imgui's internal API now) and the old C89-API has a new mangling style.

Reply:
> That, however, raises the earlier problem of ming being unhappy for Windows cross compilation.

This may be fixed by the linker flags that I mentioned in [this comment](https://github.com/JuliaPackaging/Yggdrasil/pull/1054#issuecomment-631218233). Even if it won't work, we can make a pure libimgui-cpp shared library in CImGui_jll and let both libcimgui and libcimplot link to it.

Reply:
Here's a demo that can display both the ImPlot demo window and a manually created sin curve plot. I included a CMakeLists.txt for building the binary. You'll need the libcimgui.so binary from `CImGui_jll.jl` v1.75 along with matching imgui source + cimplot/implot all in the source dir.
https://gist.github.com/wsphillips/ce974be887af312ddf2b0de0f0f30b43

Reply:
@Gnimuc I'm attempting to work with the linker flags you suggested. Should these be wrapped in an if block for Windows only? Right now if I add them to the build flags ld tells me `unrecognized option '--export-all-symbols'` but `ld --help | grep export-all-symbols` says it does indeed exist ("to DLL"). If I use mingw64 as the default compiler and build, it instead segfaults during the linking step (with flags enabled) :sweat_smile: